src/parser/postgresparser.cpp

```diff
@@ -751,42 +749,45 @@ std::unique_ptr<SelectStatement> PostgresParser::SelectTransform(SelectStmt *roo
       }
       auto limit_desc = std::make_unique<LimitDescription>(limit, offset);
 
-      result = std::make_unique<SelectStatement>(std::move(target), select_distinct, std::move(from), std::move(where),
+      result = std::make_unique<SelectStatement>(std::move(target), select_distinct, std::move(from), where,
                                                  std::move(groupby), std::move(orderby), std::move(limit_desc));
       break;
     }
     case SETOP_UNION: {
-      result = SelectTransform(root->larg_);
-      result->SetUnionSelect(SelectTransform(root->rarg_));
+      result = SelectTransform(parse_result, root->larg_);
+      result->SetUnionSelect(SelectTransform(parse_result, root->rarg_));
       break;
     }
     default: {
-      // TODO(Wan): is Set the right message, or should it be Select?
-      PARSER_LOG_AND_THROW("SelectTransform", "Set operation", root->type_);
+      PARSER_LOG_AND_THROW("SelectTransform", "Select operation", root->type_);
     }
   }
 
   return result;
 }
 
 // Postgres.SelectStmt.targetList -> terrier.SelectStatement.select_
-std::vector<std::shared_ptr<AbstractExpression>> PostgresParser::TargetTransform(List *root) {
+std::vector<common::ManagedPointer<AbstractExpression>> PostgresParser::TargetTransform(ParseResult *parse_result,
```

Comment:
Why is this different from the other transforms? All the other transforms return unique pointers

Reply:
This is one of the helpers for creating a select statement. If you see the header file, the top level SQLStatements get unique ptrs, the helper and stuff-in-them-to-abstract-expression are managed ptrs.